### PR TITLE
Traceability and organisation identifier statistics

### PR DESCRIPTION
**Pull request by radix0000**
Merge traceability and organisation identifier stats into main branch

### CODE REVIEW

File: stats/analytics.py [hunk 1]
```diff
@@ -149,6 +151,45 @@ def get_codelist_mapping(major_version):
         )
 
 
+def build_org_prefix_list():
+    """Build lists of valid organisation identifier prefixes"""
+    out = {}
+    for major_version in ('1', '2'):
+        out[major_version] = defaultdict(list)
+        for prefix in CODELISTS[major_version]['OrganisationRegistrationAgency']:
+            out[major_version][len(prefix)].append(prefix)
```
Comment (andylolz) [inline stats/analytics.py:160]: What’s the thinking behind splitting the list by prefix length? Why not just make a big list of prefixes by version?

I think it’s okay either way, I’m mostly just curious.

File: stats/analytics.py [hunk 2]
```diff
@@ -149,6 +151,45 @@ def get_codelist_mapping(major_version):
...
+org_prefix_list = build_org_prefix_list()
+
+
+def build_channel_code_list():
+    """Build lists of CRS Channel Codes"""
+    out = {}
+    for major_version in ('1', '2'):
+        out[major_version] = defaultdict(list)
+        for code in CODELISTS[major_version]['CRSChannelCode']:
+            out[major_version][code[:2]].append(code)
```
Comment (andylolz) [inline stats/analytics.py:173]: Same sort of question [as above](https://github.com/codeforIATI/IATI-Stats/pull/34/files#r1010363895) – why group by the first bit of the channel code?

**Comment by andylolz** [comment]
I’ll look in a tiny bit more detail, but from what I’ve seen at so far this looks \*great\*.

It looks like you’re maybe using `black` which is much appreciated. Similarly, the tests added here are very much appreciated. 🙏🏻

**Comment by radix0000** [comment]
Re prefix list thing, when initially started wasn't sure how large the prefix lists were going be etc, and so was thinking about ways to not have to loop over all them a million times. As it turned out doesn't seem to been an issue, but that worry is why it ended up structured like that.

**Review by markbrough — APPROVED** [review]
This is great, thank you! We will probably want to adjust some of this later but definitely good to go for now!

**Pull request merged**